src/lib/y2users/linux/local_reader.rb

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) [2021] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2users/parsers/group"
+require "y2users/parsers/passwd"
+require "y2users/parsers/shadow"
+
+module Y2Users
+  module Linux
+    # Reads local users configuration from the system using /etc files.
+    class LocalReader
```

Comment:
`Linux::LocalReader` and `Linux::Reader` are quite similar. The only one difference is how they get the sources (i.e, reading a file or calling to *getent*). I would suggest to unify them (e.g., using a constructor parameter to indicate the source?) or to create a base class or something that avoids the repetition in the code.

Reply:
Well, it start differs as Linux::Reader will call LocalReader to get source of users/groups/etc. I just do not write it yet